driver-core/src/main/com/mongodb/internal/logging/StructuredLogger.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.internal.logging;
+
+import com.mongodb.connection.ClusterId;
+import com.mongodb.internal.VisibleForTesting;
+import com.mongodb.internal.diagnostics.logging.Logger;
+import com.mongodb.internal.diagnostics.logging.Loggers;
+import com.mongodb.internal.logging.StructuredLogMessage.Entry;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.mongodb.internal.VisibleForTesting.AccessModifier.PRIVATE;
+import static java.lang.String.format;
+
+/**
+ * <p>This class is not part of the public API and may be removed or changed at any time</p>
+ */
+public class StructuredLogger {
+
+    private final Logger logger;
+
+    private static final Map<String, StructuredLoggingInterceptor> INTERCEPTORS = new HashMap<>();
+
+    public static void addInterceptor(final String applicationName, final StructuredLoggingInterceptor interceptor) {
+        INTERCEPTORS.put(applicationName, interceptor);
+    }
+
+    public static void removeInterceptor(final String applicationName) {
+        INTERCEPTORS.remove(applicationName);
+    }
+
+    public static boolean hasInterceptor(final ClusterId clusterId) {
+        return INTERCEPTORS.containsKey(clusterId.getDescription());
+    }
+
+    public static StructuredLoggingInterceptor getInterceptor(final ClusterId clusterId) {
+        return INTERCEPTORS.get(clusterId.getDescription());
+    }
+
+    public StructuredLogger(final String suffix) {
+        this(Loggers.getLogger(suffix));
+    }
+
+    @VisibleForTesting(otherwise = PRIVATE)
+    public StructuredLogger(final Logger logger) {
+        this.logger = logger;
+    }
+
+    public boolean isDebugEnabled(final ClusterId clusterId) {
+        return logger.isDebugEnabled() || hasInterceptor(clusterId);
+    }
+
+    public void debug(final String message, final ClusterId clusterId, final String format,
+            final String k1, final Object v1,
+            final String k2, final Object v2,
+            final String k3, final Object v3,
+            final String k4, final Object v4,
+            final String k5, final Object v5,
+            final String k6, final Object v6) {
+        if (hasInterceptor(clusterId)) {
```

Comment:
How expensive is this check?

Reply:
Looking at HashMap source code, looks like it amounts to a null check on `HashMap.table`